source/installation_and_configuration/authentication/fireedge.rst

```diff
@@ -0,0 +1,95 @@
+
+* **Web client and Fireedge server**. Authentication is based on the credentials stored in the OpenNebula database for the user. Depending on the type of these credentials the authentication method can be: ``remote``or ``opennebula``
+
+The following sections explain the client-to-Sunstone server authentication methods.
+
+Basic Auth
+===========
+
+In the basic mode, username and password are matched to those in OpenNebula's database in order to authorize the user at the time of login. JWT-based sessions are then used to authenticate and authorize the requests.
+
+.. code-block:: yaml
+
+    :auth: opennebula
+
+Remote Auth
+===========
+
+This method performs OpenNebula login based on user extraction and compares it with the password value in the user database.
+
+To update existing users to use remote authentication, change the controller to public and update the password as follows:
+
+The user password has to be changed by running one of the following commands:
+
+.. prompt:: bash $ auto
+
+    $ oneuser chauth johndoe public "johndoe"
+
+New users with this authentication method should be created as follows:
+
+.. prompt:: bash $ auto
+
+    $ oneuser create johndoe "johndoe" --driver public
+
+To enable this login method, set the ``:auth:`` option in ``/etc/one/sunstone-server.conf`` to ``remote`` and restart Sunstone:
+
+.. code-block:: yaml
+
+    :auth: remote
+
+The login screen will not display the username and password fields anymore, as all information is fetched from the user certificate:
+
+|fireedge_remote_login|
+
+Note that OpenNebula will not verify that the user holds a valid certificate at the time of login: this is expected to be done by the external container of the Fireedge server (normally Apache), whose job is to tell the user's browser that the site requires a user certificate and to check that the certificate is consistently signed by the chosen Certificate Authority (CA).
+
+.. warning:: The Fireedge authentication only handles the authentication of the user at the time of login. Authentication of the user certificate is a complementary setup, which can rely on Apache.
+
+.. _2f_auth_fireedge:
+
+Two Factor Authentication
+=========================
+
+You can get an additional authentication level by using a two-factor authentication that not only requests the username and password but also the one-time (or pre-generated security) keys generated by an authenticator application.
+
+|fireedge_2fa_auth|
+
+Authenticator App
+------------------
+
+This method requires a token generated by any of these applications: `Google Authentication <https://play.google.com/store/apps/details?id=com.google.android.apps.authenticator2&hl=en>`__, `Authy <https://authy.com/download/>`__ or `Microsoft Authentication <https://www.microsoft.com/en-us/p/microsoft-authenticator/9nblgggzmcj6?activetab=pivot:overviewtab>`__.
```

Comment:
is this working in Fireedge? i thought we were waiting for the user tab to implement all this

Reply:
AFAIK, 2FA works in FireEdge as it asks for the code for logging however, it can't be configured on FireEdge Sunstone